src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jobs/oneoff/migration/ChangelogMigrationProcessor.kt

```diff
@@ -0,0 +1,83 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jobs.oneoff.migration
+
+import mu.KLogging
+import org.springframework.batch.item.ItemProcessor
+import org.springframework.stereotype.Component
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto.ReferralAmendmentDetails
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.Changelog
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.entity.ReferralDetails
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.AuthUserRepository
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.ChangelogRepository
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.jpa.repository.ReferralDetailsRepository
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service.AmendTopic
+import java.util.UUID
+
+@Component
+class ChangelogMigrationProcessor(
+  private val changelogRepository: ChangelogRepository,
+  private val referralDetailsRepository: ReferralDetailsRepository,
+  private val authUserRepository: AuthUserRepository
+) : ItemProcessor<ReferralDetails, Changelog?> {
+  companion object : KLogging()
+
+  override fun process(referralDetails: ReferralDetails): Changelog? {
+    logger.info("processing referral details {} for referral id {}", referralDetails.id, referralDetails.referralId)
+    // check if any changlog is null
+    if (referralDetails.supersededById != null) {
+      return try {
+        // Do this migration when there is no change log data
+        migrateReferralToChangelogTable(referralDetails)
+      } catch (e: Exception) {
+        // log and continue
+        logger.info("logging the referral details that failed migration {}, {}", referralDetails.id, e.message)
```

Comment:
Can we let the exception bubble up? I'm not sure `info` would be visible to our monitoring and we lose the original stacktrace this way

Reply:
thrown the exception